src/Storages/StorageMerge.cpp

```diff
@@ -275,6 +276,12 @@ void StorageMerge::read(
     StorageListWithLocks selected_tables
         = getSelectedTables(modified_context, query_info.query, has_database_virtual_column, has_table_virtual_column);
 
+    Strings tables;
+    tables.reserve(selected_tables.size())
+    for (const auto & entry : selected_tables)
+        tables.push_back(fmt::format("{}.{}", std::get<0>(entry), std::get<3>(entry)));
+    LOG_TRACE(log, "Will read from {} tables: ", fmt::join(tables, ","));
```

Comment:
It can be dangerous, as in [Metrica](https://metrica.yandex.com/), they have a `Merge` table over 1 000 000 `StripeLog` tables on a cluster for intermediate data processing. So, every query to this table will write tens of megabytes in logs.

Please add some thresholds.